src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -164,7 +164,8 @@ void remove_redundant_reorders::run(program& p) {
         bool remove_current = r_node.is_simple_reorder() &&
                               r_dep_node.get_users().size() == 1 &&
                               !r_dep_node.is_output() &&
-                              !r_node.get_primitive()->has_surface_input();
+                              !r_node.get_primitive()->has_surface_input() &&
+                              r_node.get_input_layout().data_padding == r_node.get_output_layout().data_padding;
```

Comment:
As this is about shink of reorder "chain", I expect we can handle different data padding.

Reply:
I agree. @kelvinchoi-intel  Could you check quickly whether it works with adding merge_output_padding below ?

Reply:
Yes, your suggestion would be better way but, I'm not sure the impact on other models. So I've fix to least impact way. My team also discussed about the concern and decided to consider adding propagate padding when we faced the related issue later.
For the reported model, tiny sd unet, the issued reorder layer is removed as propagating padding info on rest remove redundant process.

Reply:
@kelvinchoi-intel I guess that your comment of "My team discussed..." indicates the discussion with me...?